### PR TITLE
fix(paths): return json for path data

### LINKED ISSUE #49: missing path data
Not sure if something is breaking in the build for path data, but there seems to be various icons that [export an empty string](https://github.com/planningcenter/icons/blob/master/paths/check-ins.js#L17).

### PR DESCRIPTION
**Pull request by souporserious**
This PR replaces `extract-svg-path` with `svgson` in order to preserve all of the children in an SVG. It looks like `extract-svg-path` assumes everything is made up of paths which messes up some of the exports. I tried to normalize the data by just returning the children responsible for the graphic, but it might be worth it to just include all of the JSON. Also, the "path" naming may be a little inflated now since it includes more than before so I'd be up for renaming to something like `data` / `json`, just let me know.

closes #49

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@planningcenter/icons",
-  "version": "7.17.2",
+  "version": "7.17.3",
```
Comment (joshwilkerson) [inline package.json:3]: @souporserious I bumped the version to `7.17.3` yesterday... https://github.com/planningcenter/icons/releases/tag/v7.17.3

Comment (souporserious) [reply package.json:3]: Ah, thanks for the heads up!

Comment (souporserious) [reply package.json:3]: I'm actually wondering if this should be considered a major change since the data is changing 🤔

**Comment by chantastic** [comment]
Thanks @souporserious for proposing a JSON option.

At the moment the icons are intentionally flat. All icon sets should be flattened to single paths.
If not, the bug is in the icon set.

The `paths` were provided as an experiment. If their in use now, should start looking into adding them to the visual documentation and testing the output.
Would love to get these into Chromatic but something local could provide a stop gap.

**Comment by souporserious** [comment]
@chantastic I'm not sure I follow, it looks like the SVGs in use on the demo site aren't using flat paths:

image

What is the particular issue with returning JSON data? 

Also, is there any mention of paths being experimental? We've been directing UI kit users to use this for the last couple of months and when I filed an issue about it originally there was no feedback.

**Comment by chantastic** [comment]
correct. these are considered bugs in the source. we've fixed these artboards by flattening them.

it's good for me to know that the flat export artboards expectation is not clear to newcomers. there haven't been newcomers for a while and with most regular contributors have internalized these types fixes.

issue #49 was missed. sorry for that. feel free to ping again or mention @dsecrest or I specifically.
the recommendation would have been the same, flattening the art and re-exporting.

the functionality was added for Church Center App and that's the only regular consumer of paths that i was aware of. i wasn't told that UI-Kit was recommending the paths.

with regard to representing the glifs as json:
i don't see any issue with the technique. but, as i mentioned, the expectation is that icons be flattened by their designers. there have been a number of unpredictable artifacts with different blend modes passed thru svgo. because none of the current consumers target individual shapes, there's no case to take on the complexity.

however, i do see improving our testing — mentioned in the original reply — as the first step toward the confidence we'd need in making such a change.

i hope this provides the context you're looking for. i will add the flattening information to docs.

**Comment by tage** [comment]
Thanks for taking time to write this up @souporserious!

So If I'm reading this correctly @chantastic, the current flattened path workflow was added originally for Church Center App. Is this the main reason for the path data requirement? Asking out of curiosity.

Our icons are very simple and we don't currently have a need for anything but flattened icons, but if we did accept and interpret any kind of SVG data, it _seems_ like we could eliminate bugs on export from design tools?

I don't see a compelling reason to tackle this now, but it seems like a conversation we could maybe reopen in the future? Regardless, this should be for the fidelity team to determine.

**Comment by chantastic** [comment]
@tage flattened dartboards has been a recommendation from the first version of Icons.

it continued after transitioning icons to SVG Sprites because they are naturally less capable than direct SVG usage. for example, you can't target specific shapes for coloring or transforming.

the rule is a "soft rule" in the sense that SVG Sprites won't "break" if ignored.
however, sometimes tooling changes can cause interruptions with paths aren't blended by the designer in design tools.
so, we keep the rule to ensure that designers control the final output.

the ability to use `paths` as a string was added to support Church Center App and the icon sets that they use.
the use of paths does provide a dead simple implementation product-side.

for projects as widely consumed (and maintained) as Icons, we try to keep suitability and capability closely aligned.
without an immediate or prolonged allow for multiple paths, adding it is like building a API around a bug — with the added potential to add drift in expectations between SVG Sprite and path consumers.

as mentioned before, the path forward will be to get all manifestations of the icons under test, to ensure that nothing is broken when exploring other ideas.
after that, i'd be interested in seeing what features actually require multiple paths and if they justify the additional, suite-wide, complexity.

**Comment by dannydb** [comment]
I’m open to the JSON solution, but as @chantastic mentions, we don’t necessarily _need_ that flexibility and it will require changes to the CCA icon implementation. I’ve run into the issue of non-flattened paths in non-`church-center` sets as we’ve transitioned to using the `ui-kit` icon component as a consumer of the path export, but I was already aware of the flat path requirement and fixing those instances was not a huge deal.

**Comment by chantastic** [comment]
I think that we're going to resolve this with a few R&F cards to flatten the problematic icons and — in the case of `check-ins#delete` rename the icon. Will have all those ready by the kick-off meeting 9/14.

I'll close this once those cards are up.

**Pull request closed without merging**